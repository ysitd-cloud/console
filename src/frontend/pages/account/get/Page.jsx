import React from 'react';
import { graphql } from 'react-apollo';
import query from './query.graphql';
import dataProps from '../../../common/react-apollo-props';
import ProgressLinear from '../../../components/react/ProgressLinear';
import Flex from '../../../components/react/Flex';
import UserCard from './UserCard';

export function Page({ data }) {
  return (
    <Flex className="xs12">
      {data.loading ? <ProgressLinear indeterminate /> : <UserCard
        username={data.user.username}
        avatarUrl={data.user.avatarUrl}
        displayName={data.user.displayName}
      />}
    </Flex>
  );
}

Page.propTypes = {
  data: dataProps,
};

Page.defaultProps = {
  data: null,
};

const PageWithQuery = graphql(query, {
  options: ({ username }) => ({
    variables: { username },
    fetchPolicy: 'cache-and-network',
  }),
})(Page);

export default PageWithQuery;
