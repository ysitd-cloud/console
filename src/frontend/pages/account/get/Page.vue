<template>
    <v-flex xs12>
        <user-card v-if="ready" :avatarUrl="user.avatarUrl" :name="user.displayName" :username="user.username"></user-card>
        <v-progress-linear color="primary" v-else-if="!ready && !error" indeterminate></v-progress-linear>
        <error-card v-else-if="error !== null" :error="error"></error-card>
    </v-flex>
</template>

<script src="./script.js"></script>