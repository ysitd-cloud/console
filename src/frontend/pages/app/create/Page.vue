<template>
    <v-flex xs12>
        <v-form @submit="() => false">
            <div class="headline mt-2 mb-3 mx-1">Create Application</div>

            <v-card class="mb-2">
                <v-card-title class="pb-0">
                    <div>Metadata</div>
                </v-card-title>
                <v-card-text class="pt-0">
                    <v-layout row>
                        <v-flex xs12 md8>
                            <v-text-field label="Name" id="name" v-model="app.name" required></v-text-field>
                        </v-flex>
                    </v-layout>
                </v-card-text>
            </v-card>

            <v-card class="mb-2">
                <v-card-title class="pb-0">
                    <div>Deployment</div>
                </v-card-title>
                <v-card-text class="pt-0">
                    <v-layout row>
                        <v-flex xs12 md6 lg3 xl2>
                            <v-text-field label="Image" id="image" v-model="app.deployment.image" required></v-text-field>
                        </v-flex>
                        <v-flex xs12 md6 lg3 xl2 class="ml-1">
                            <v-text-field label="Tag" id="tag" v-model="app.deployment.tag" required></v-text-field>
                        </v-flex>
                    </v-layout>
                </v-card-text>
            </v-card>

            <v-card class="mb-2">
                <v-card-title class="pb-0">
                    <div>Network</div>
                </v-card-title>
                <v-card-text class="pt-0">
                    <v-layout row>
                        <v-flex xs12 md8>
                            <v-text-field label="Domain" id="domain" v-model="app.network.domain" required></v-text-field>
                        </v-flex>
                    </v-layout>
                </v-card-text>
            </v-card>

            <v-card class="mb-2">
                <v-card-title class="pb-0">
                    <div>Environment</div>
                </v-card-title>
                <v-card-text class="py-0">
                    <v-layout row v-for="(env, i) in app.environment" :key="i">
                        <v-flex xs12 md6 lg3 xl2>
                            <v-text-field label="Name" v-model="env.key" required></v-text-field>
                        </v-flex>
                        <v-flex xs12 md6 lg3 xl2 class="ml-1">
                            <v-text-field label="Value" v-model="env.value" required></v-text-field>
                        </v-flex>
                        <v-btn icon class="red--text mb-0 mt-3" @click="() => removeEnv(i)">
                            <v-icon>remove_circle_outline</v-icon>
                        </v-btn>
                    </v-layout>
                </v-card-text>
                <v-card-actions class="py-0">
                    <v-btn icon class="green--text" @click="addEnv">
                        <v-icon>add_circle_outline</v-icon>
                    </v-btn>
                </v-card-actions>
            </v-card>

            <v-layout row>
                <v-flex xs12>
                    <v-btn color="primary" type="submit">
                        <v-icon left>unarchive</v-icon>
                        Deploy
                    </v-btn>
                </v-flex>
            </v-layout>
        </v-form>
    </v-flex>
</template>

<script src="./script.js"></script>

<!--<style scoped lang="stylus" ></style>-->