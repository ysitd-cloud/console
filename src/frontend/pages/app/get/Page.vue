<template>
    <v-flex xs12>
        <v-btn color="info" :to="{ name: 'app.list' }">
            <v-icon left>keyboard_arrow_left</v-icon>Back
        </v-btn>
        <v-btn @click="loadApp" :disabled="!ready" v-if="!error">
            <v-icon>cached</v-icon>Refresh
        </v-btn>
        <v-card v-if="ready && !error && app != null">
            <v-card-title>
                <h3 class="mb-0">{{ app.name }}</h3>
            </v-card-title>
            <v-divider></v-divider>
            <v-card-text class="pb-0">
                <p>
                    Image: <v-chip>{{ app.deployment.image }}:{{ app.deployment.tag }}</v-chip>
                </p>
                <p>
                    Network: <a :href="`http://${app.network.domain}`" target="_blank">{{ app.network.domain }}</a>
                </p>
            </v-card-text>
            <div class="pl-3">Environment</div>
            <v-data-table :headers="[ { text: 'Name', value: 'key', align: 'left' }, { text: 'Value', value: 'value', align: 'left' } ]" :items="app.environment" hide-actions>
                <template slot="items" slot-scope="props">
                    <td>{{ props.item.key }}</td>
                    <td>
                        <hidable-text :content="props.item.value"></hidable-text>
                    </td>
                </template>
            </v-data-table>
        </v-card>
        <error-card v-if="error" :error="error"></error-card>
        <v-progress-linear v-if="!ready && !error" indeterminate></v-progress-linear>
    </v-flex>
</template>

<script src="./script.js"></script>
<style scoped src="./style.styl" lang="stylus"></style>
