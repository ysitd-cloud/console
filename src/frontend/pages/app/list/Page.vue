<template>
    <v-flex xs12>
        <v-card v-if="ready">
            <v-card-title>
                <h3 class="mb-0">App</h3>
            </v-card-title>
            <v-list v-if="apps.length > 0">
                <v-list-tile v-for="app in apps" :key="app.id">
                    <v-list-tile-content>
                        <v-list-tile-title>{{ app.name }}</v-list-tile-title>
                    </v-list-tile-content>
                </v-list-tile>
            </v-list>
            <v-card-text v-else class="ml-2 mb-2">
                No app is created. Go and create one.
            </v-card-text>
        </v-card>
        <error-card v-if="error" :error="error"></error-card>
        <v-progress-linear v-if="!ready && !error" indeterminate></v-progress-linear>
    </v-flex>
</template>

<script src="./script.js"></script>
