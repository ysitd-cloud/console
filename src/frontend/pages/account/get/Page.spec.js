import React from 'react';
import renderer from 'react-test-renderer';
import { Page } from './Page';

/* globals describe, expect, it, jest */

describe('React Card Component', () => {
  it('loading match snapshot', () => {
    const data = {
      loading: true,
      error: null,
      refetch: jest.fn(),
      fetchMore: jest.fn(),
      updateQuery: jest.fn(),
    };
    const component = renderer.create(<Page data={data} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('ready match snapshot', () => {
    const data = {
      loading: false,
      error: null,
      refetch: jest.fn(),
      fetchMore: jest.fn(),
      updateQuery: jest.fn(),
      user: {
        username: 'foobar',
        displayName: 'foo',
        avatarUrl: 'bar.jpg',
      },
    };
    const component = renderer.create(<Page data={data} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
