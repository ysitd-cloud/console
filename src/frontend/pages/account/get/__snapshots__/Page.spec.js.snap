// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Card Component loading match snapshot 1`] = `
<div
  className="flex xs12"
  style={Object {}}
>
  <div
    className="progress-linear__bar"
    style={
      Object {
        "height": "7px",
      }
    }
  >
    <div
      className="primary progress-linear__background"
      style={
        Object {
          "height": "auto",
          "opacity": 0.3,
          "width": "100%",
        }
      }
    />
    <div
      className="progress-linear__bar"
      style={Object {}}
    >
      <div
        className="progress-linear__bar__indeterminate progress-linear__bar__indeterminate--active"
      >
        <div
          className="progress-linear__bar__indeterminate long primary"
        />
        <div
          className="progress-linear__bar__indeterminate short primary"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`React Card Component ready match snapshot 1`] = `
<div
  className="flex xs12"
  style={Object {}}
>
  <div
    className="card"
    style={
      Object {
        "height": "auto",
      }
    }
  >
    <div
      className="layout row"
    >
      <div
        className="flex xs12 md2 lg1"
        style={Object {}}
      >
        <img
          alt="Avatar Icon"
          src="bar.jpg"
          style={
            Object {
              "minHeight": "50px",
            }
          }
        />
      </div>
      <div
        className="flex xs12 md8 lg11"
        style={Object {}}
      >
        <div
          className="card__title pb-0"
        >
          <div
            className="headline mb-0"
          >
            <span
              className="bold"
            >
              Name: 
            </span>
            foo
          </div>
        </div>
        <div
          className="card__text pt-1"
        >
          <div>
            <span
              className="bold"
            >
              Username: 
            </span>
            foobar
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
